File: Java/0007_Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        int reverse = 0;

        // for all digits in x, pop last digit(tail) and push it to the back of the reverse
        while (x != 0) {
            int tail = x % 10;                              // pop last digit
            int tempReverse = reverse * 10 + tail;          // push it to the back of the reverse
            if ((tempReverse - tail) / 10 != reverse) {     // check overflows
                return 0;
            }
            reverse = tempReverse;
            x = x / 10;                                     // remove last digit
        }
        return reverse;
    }
}

File: JavaScript/0007_Reverse_Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    // x = -123
    // 1) get absolute value of x,  x = 123
    // 2) convert x to string,      x = '123'
    // 3) split x,                  x = [1,2,3]
    // 4) revese an array x,        x = [3,2,1]
    // 5) concatenate all elements, x = '321'
    const reversed = Math.abs(x).toString().split('').reverse().join('');

    // check overflows
    if (reversed > 2**31) {
        return 0;
    }

    // return with sign
    return reversed * Math.sign(x);
};
